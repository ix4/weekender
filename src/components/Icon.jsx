import React from 'react'
import PropTypes from 'prop-types'

import ICON_AIGA_BUS from '../images/aiga_bus_clear.svg'
import INTERNATIONAL_SYMBOL_ACCESSIBILITY from '../images/isa.svg'

const Icon = (props) => {
  switch (props.type) {
    case 'bus':
      return <img src={ICON_AIGA_BUS} className="subway-bullet-small" alt="[bus icon]" />
    case 'isa':
      return <img src={INTERNATIONAL_SYMBOL_ACCESSIBILITY} className="icon-isa" alt="[International Symbol of Accessibility]" />
    default:
      break
  }

  return null
}

Icon.prototype.propTypes = {
  type: PropTypes.string.isRequired
}

export default Icon
