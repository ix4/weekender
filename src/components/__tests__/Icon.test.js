/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import Icon from '../Icon'

describe('<Icon>', () => {
  it('renders bus icon', () => {
    const component = renderer.create(
      <Icon type="bus" />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()  
  })

  it('renders accessibility icon', () => {
    const component = renderer.create(
      <Icon type="isa" />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()  
  })
})
