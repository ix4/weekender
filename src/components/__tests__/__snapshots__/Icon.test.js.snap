// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon> renders accessibility icon 1`] = `
<img
  alt="[International Symbol of Accessibility]"
  className="icon-isa"
  src="test-file-stub"
/>
`;

exports[`<Icon> renders bus icon 1`] = `
<img
  alt="[bus icon]"
  className="subway-bullet-small"
  src="test-file-stub"
/>
`;
